tests/test_binary.py

```diff
@@ -0,0 +1,113 @@
+logger.setLevel(logging.DEBUG)
+
+BINARY_DISABLE_WEB_OPTION = "-nw"
+LOG_CONTENT_TEST_WAITING_TIME = 120
+
+
+@pytest.fixture
+def start_binary():
+    clear_octobot_previous_folders()
+    with TemporaryFile() as output, TemporaryFile() as err:
+        binary_process = create_binary("", output, err)
+        yield
+        terminate_binary(binary_process, output, err)
+
+
+@pytest.fixture
+def start_binary_without_web_app():
+    clear_octobot_previous_folders()
+    with TemporaryFile() as output, TemporaryFile() as err:
+        binary_process = create_binary(BINARY_DISABLE_WEB_OPTION, output, err)
+        logger.debug(err.read())
+        yield
+        terminate_binary(binary_process, output, err)
+
+
+def create_binary(binary_options, output_file, err_file):
+    logger.debug("Starting binary process...")
+    return subprocess.Popen(f"{get_binary_file_path()} {binary_options}",
+                            shell=True,
+                            stdout=output_file,
+                            stderr=err_file,
+                            preexec_fn=os.setsid if not is_on_windows() else None)
+
+
+def terminate_binary(binary_process, output_file, err_file):
+    logger.info(output_file.read())
+    errors = err_file.read()
+    if errors:
+        logger.error(errors)
+        raise ValueError(f"Error happened during process execution : {errors}")
+    logger.debug("Killing binary process...")
+    if is_on_windows():
+        os.kill(binary_process.pid, signal.CTRL_C_EVENT)
+    else:
+        try:
+            os.killpg(os.getpgid(binary_process.pid), signal.SIGTERM)  # Send the signal to all the process groups
+        except ProcessLookupError:
+            binary_process.kill()
+
+
+def test_version_endpoint(start_binary):
+    max_attempts = 10
+    attempt = 1
+    while max_attempts >= attempt > 0:
+        try:
+            requests.get('http://localhost:5001/version')
+            attempt = -1  # success
+        except requests.exceptions.ConnectionError:
+            logger.warning(f"Failed to get http://localhost/version, retrying ({attempt}/{max_attempts})...")
+            attempt += 1
+            time.sleep(1)
+    log_content = get_log_file_content()
+    logger.debug(log_content)
+    assert attempt <= max_attempts
+
+
+def test_evaluation_state_created(start_binary_without_web_app):
+    time.sleep(LOG_CONTENT_TEST_WAITING_TIME)
+    log_content = get_log_file_content()
+    logger.debug(log_content)
+    assert "new state:" in log_content
+
+
+def test_logs_content_has_no_errors(start_binary_without_web_app):
+    time.sleep(LOG_CONTENT_TEST_WAITING_TIME)
+    log_content = get_log_file_content()
+    logger.debug(log_content)
+    assert "ERROR" not in log_content
+
+
+def test_balance_profitability_updated(start_binary_without_web_app):
+    time.sleep(LOG_CONTENT_TEST_WAITING_TIME)
+    log_content = get_log_file_content()
+    logger.debug(log_content)
+    assert "BALANCE PROFITABILITY :" in log_content
```

Comment:
great tests !

Reply:
Maybe we could add while loop for the `test_version_endpoint`, `test_evaluation_state_created` and `test_balance_profitability_updated` instead of time.sleep() to speed up the test when possible ?

Reply:
I don't think that there is a lot of time to save, but it would be a great improvement.